src/run_cache_sim.cpp

```diff
@@ -30,6 +31,11 @@ int main(int argc, char *argv[]) {
     return 1;
   }
 
+  if (sample_n_lines > n_lines || sample_n_lines == -1) {
+    sample_n_lines = n_lines;
+    std::cerr << "Target samples too big... max to file size" << "\n";	
```

Comment:
"sim_first_n_lines must be less than or equal to the number of lines in the file"

Reply:
sample_n_lines == -1 is acceptable, right? isn't that the default? also maybe you should return -1; here since this is an incorrect input

Reply:
ah this is to handle default, or when trace file size is smaller than I thought

Reply:
ack- we shouldn't print an error though